docs/DeveloperGuide.md

```diff
@@ -346,40 +346,590 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+`* * *` | user | view all tasks under a specific list | view my tasks by categories
+`* * *` | user | create a customised list other than the default lists | personalize my categories
+`* * *` | user | delete task(s) | remove task(s) that is/are no longer useful or created by mistake
+`* * *` | user | view the help message for a specific feature | know how to use a command
+`* * *` | first-time user | view the entire help messages | get to know various commands
+`* *` | user | view next n days tasks | know what to do in n days' time
+`* *` | user | update the name/time/due date/description/priority level/... of task | change its content
+`* *` | user | add a task which repeats | save the effort of creating it every week/day
+`* *` | user | view the list of tasks by due date| view the tasks due on that day
+`* *` | user | specify priority level for a task when adding the task| know which task is more urgent
+`* *` | user | undo my latest command| undo when I regret my latest action
+`* *` | user | delete one most recent occurrence of a repeating task | choose not to see the task for one time
+`*` | user | add in the map of the venue | know how to get to a venue
+`*` | user | add the venue of the task | add a venue to the task and link it to the venue on google map
+`*` | user | redo my undone command| redo when I regret my undo command
+`*` | user | view the finished tasks under each list | see what I have already completed
+`*` | user | delete the list and put all the tasks inside into Personal or delete all the tasks | delete the list and delete tasks by lists
+`*` | user | see different colours for different priority levels | know which one is more urgent
+`*` | user | specify priority level for a task after adding the task | know which task is more urgent
+`*` | user | mark a task to be my favourite | view it under list favorite 
+`*` | user | update the name of the list | modify it after creating
+`*` | user | view all the starred tasks using a "favourite" command | view all my favourite tasks
+`*` | user | see the autocomplete messages when I type | know what I can type
+`*` | advanced user | change the appearance of the user interface | personalize the view
+`*` | user | view all the past due tasks | get to know what task is left undone
+`*` | user | specify its list when adding a task | group tasks that belong to the same category
+`*` | user | list all list names | get to know current lists
+`*` | user | view finished tasks under current/another list | view my completed tasks
+`*` | user | view/list all finished and unfinised tasks | manage all tasks
 
 {More to be added}
 
 ## Appendix B : Use Cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `Dueue` and the **Actor** is the `user`, unless specified otherwise)
 
-#### Use case: Delete person
+#### Use case: Add task by name only
```

Comment:
Should consider use case that are more challenging :-). For example, to delete task, you probably need to view the task to find out the index so that you can request for delete.